### ISSUE TITLE
New Windows worker node; pods have no network connectivity

### ISSUE THREAD

**Issue by liquidtoon01** [projectcalico/calico#4682]
Hi, after deploying my first Windows node to the cluster it cannot access anything other pods/services or the LAN/WAN. Pod get scheduled fine.

## Expected Behavior
Pod to Pod network connectivity works
Pod to Service connectivity works
Pod to WAN (internet) connectivity works
Pod to LAN (local intranet) connectivity works

## Current Behavior
Zero network connectivity

## Possible Solution
unknown

## Steps to Reproduce (for bugs)
Following the quickstart guide; https://docs.projectcalico.org/getting-started/windows-calico/quickstart in VXLAN mode
Schedule change service CIDR and DNS to match correctly
schedule windows pod

## Your Environment

```
PS C:\> curl bbc.co.uk 
curl : The remote name could not be resolved: 'bbc.co.uk' 
At line:1 char:1
+ curl bbc.co.uk
+ ~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebException 
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```

```
# calicoctl get ippool -o wide
NAME                  CIDR          NAT    IPIPMODE   VXLANMODE   DISABLED   SELECTOR
default-ipv4-ippool   10.1.0.0/16   true   Never      Always      false      all()
```
config.ps1
```
$baseDir = "$PSScriptRoot"
ipmo $baseDir\libs\calico\calico.psm1

## Cluster configuration:

# KUBE_NETWORK should be set to a regular expression that matches the HNS network(s) used for pods.
# The default, "Calico.*", is correct for Calico CNI. 
$env:KUBE_NETWORK = "Calico.*"

# Set this to one of the following values:
# - "vxlan" for Calico VXLAN networking
# - "windows-bgp" for Calico BGP networking using the Windows BGP router.
# - "none" to disable the Calico CNI plugin (so that you can use another plugin).
$env:CALICO_NETWORKING_BACKEND="vxlan"

# Set to match your Kubernetes service CIDR.
$env:K8S_SERVICE_CIDR = "10.152.183.0/24"
$env:DNS_NAME_SERVERS = "10.152.183.10"
$env:DNS_SEARCH = "svc.cluster.local"


## Datastore configuration:

# Set this to "kubernetes" to use the kubernetes datastore, or "etcdv3" for etcd.
$env:CALICO_DATASTORE_TYPE = "kubernetes"

# Set KUBECONFIG to the path of your kubeconfig file.
$env:KUBECONFIG = "$PSScriptRoot\calico-kube-config"

# For the "etcdv3" datastore only: set ETCD_ENDPOINTS, format: "http://<host>:<port>,..."
$env:ETCD_ENDPOINTS = ""
# For etcd over TLS, set these lines to point to your keys/certs:
$env:ETCD_KEY_FILE = ""
$env:ETCD_CERT_FILE = ""
$env:ETCD_CA_CERT_FILE = ""


## CNI configuration, only used for the "vxlan" networking backends.

# Place to install the CNI plugin to.  Should match kubelet's --cni-bin-dir.
$env:CNI_BIN_DIR = "c:\k\cni"
# Place to install the CNI config to.  Should be located in kubelet's --cni-conf-dir.
$env:CNI_CONF_DIR = "c:\k\cni\config"

if (Get-IsContainerdRunning)
{
    $env:CNI_BIN_DIR = Get-ContainerdCniBinDir
    $env:CNI_CONF_DIR = Get-ContainerdCniConfDir
}

$env:CNI_CONF_FILENAME = "10-calico.conf"
# IPAM type to use with Calico's CNI plugin.  One of "calico-ipam" or "host-local".
$env:CNI_IPAM_TYPE = "calico-ipam"

## VXLAN-specific configuration.

# The VXLAN VNI / VSID.  Must match the VXLANVNI felix configuration parameter used
# for Linux nodes.
$env:VXLAN_VNI = "4096"
# Prefix used when generating MAC addresses for virtual NICs.
$env:VXLAN_MAC_PREFIX = "0E-2A"


## Node configuration.

# The NODENAME variable should be set to match the Kubernetes Node name of this host.
# The default uses this node's hostname (which is the same as kubelet).
#
# Note: on AWS, kubelet is often configured to use the internal domain name of the host rather than
# the simple hostname, for example "ip-172-16-101-135.us-west-2.compute.internal".
$env:NODENAME = $(hostname).ToLower()
# Similarly, CALICO_K8S_NODE_REF should be set to the Kubernetes Node name.  When using etcd,
# the Calico kube-controllers pod will clean up Calico node objects if the corresponding Kubernetes Node is
# cleaned up.
$env:CALICO_K8S_NODE_REF = $env:NODENAME

# The time out to wait for a valid IP of an interface to be assigned before initialising Calico
# after a reboot.
$env:STARTUP_VALID_IP_TIMEOUT = 90

# The IP of the node; the default will auto-detect a usable IP in most cases.
$env:IP = "autodetect"

## Logging.

$env:CALICO_LOG_DIR = "$PSScriptRoot\logs"

# Felix logs to screen at info level by default.  Uncomment this line to override the log
# level.  Alternatively, (if this is commented out) the log level can be controlled via
# the FelixConfiguration resource in the datastore.
# $env:FELIX_LOGSEVERITYSCREEN = "info"
# Disable logging to file by default since the service wrapper will redirect our log to file.
$env:FELIX_LOGSEVERITYFILE = "none"
# Disable syslog logging, which is not supported on Windows.
$env:FELIX_LOGSEVERITYSYS = "none"
# confd logs to screen at info level by default.  Uncomment this line to override the log
# level.
#$env:BGP_LOGSEVERITYSCREEN = "debug"

```

Service Cidr
```
# SVCRANGE=$(echo '{"apiVersion":"v1","kind":"Service","metadata":{"name":"tst"},"spec":{"clusterIP":"1.1.1.1","ports":[{"port":443}]}}' | kubectl apply -f - 2>&1 | sed 's/.*valid IPs is //')
# echo $SVCRANGE
10.152.183.0/24
```

10-coniclo.conf file
```
{
  "name": "Calico",
  "windows_use_single_network": true,

  "cniVersion": "0.3.1",
  "type": "calico",
  "mode": "vxlan",

  "vxlan_mac_prefix":  "0E-2A",
  "vxlan_vni": 4096,

  "policy": {
    "type": "k8s"
  },

  "log_level": "info",

  "capabilities": {"dns": true},

  "DNS":  {
    "Nameservers":  ["10.152.183.10"],
    "Search":  [
      "svc.cluster.local"
    ]
  },

  "nodename_file": "C:\\CalicoWindows\\libs\\calico\\..\\..\\nodename",

  "datastore_type": "kubernetes",

  "etcd_endpoints": "",
  "etcd_key_file": "",
  "etcd_cert_file": "",
  "etcd_ca_cert_file": "",

  "kubernetes": {
    "kubeconfig": "C:\\CalicoWindows\\calico-kube-config"
  },

  "ipam": {
    "type": "calico-ipam",
    "subnet": "usePodCidr"
  },

  "policies":  [
    {
      "Name":  "EndpointPolicy",
      "Value":  {
        "Type":  "OutBoundNAT",
        "ExceptionList":  [
          "10.152.183.0/24",
          "10.1.32.0/24",
          "10.1.18.0/24"
        ]
      }
    },
    {
      "Name":  "EndpointPolicy",
      "Value":  {
        "Type":  "ROUTE",
        "DestinationPrefix":  "10.152.183.0/24",
        "NeedEncap":  true
      }
    }
  ]
}
```

* Orchestrator version (e.g. kubernetes, mesos, rkt): microk8s
* Operating System and version: Windows Server 2019 1809

**Comment by lmm** [comment]
Sorry for the delay @liquidtoon01 - do you have outbound internet access on the node that is running the pod?
Could you get the output of `Get-HNSEndpoint |? ipaddress -eq <pod_ip> | ConvertTo-Json -Depth 5`?

**Comment by liquidtoon01** [comment]
Hi! Yes the node has outbound internet access. Here's the output you requested;

```
Get-HNSEndpoint |? ipaddress -eq 10.1.18.67 | ConvertTo-Json -Depth 5
{
    "ActivityId":  "CF68487D-F974-4782-8FEF-10FA99E27BF2",
    "AdditionalParams":  {

                         },
    "CreateProcessingStartTime":  132695159565404288,
    "DNSServerList":  "10.152.183.10",
    "DNSSuffix":  "default.svc.cluster.local,svc.cluster.local,cluster.local",
    "EncapOverhead":  50,
    "GatewayAddress":  "10.1.18.65",
    "Health":  {
                   "LastErrorCode":  0,
                   "LastUpdateTime":  132695159565324289
               },
    "ID":  "89958F54-AEE8-4050-B1E1-CB12A86CE8BC",
    "IPAddress":  "10.1.18.67",
    "MacAddress":  "0E-2A-0a-01-12-43",
    "Name":  "0dd146321a68d26b7d32e1bdf87ed5967296e58d97918858fb27c2ca55e3b586_Calico",
    "Policies":  [
                     {
                         "ExceptionList":  [
                                               "10.152.183.0/24",
                                               "10.1.0.0/16"
                                           ],
                         "Type":  "OutBoundNAT"
                     },
                     {
                         "DestinationPrefix":  "10.152.183.0/24",
                         "NeedEncap":  true,
                         "Type":  "ROUTE"
                     },
                     {
                         "PA":  "172.16.0.65",
                         "Type":  "PA"
                     },
                     {
                         "Action":  "Allow",
                         "Direction":  "In",
                         "Id":  "allow-host-to-endpoint",
                         "InternalPort":  0,
                         "LocalAddresses":  "",
                         "LocalPort":  0,
                         "Priority":  900,
                         "Protocol":  256,
                         "RemoteAddresses":  "172.16.0.65/32,172.26.160.1/32",
                         "RemotePort":  0,
                         "RuleType":  "Switch",
                         "Scope":  0,
                         "ServiceName":  "",
                         "Type":  "ACL"
                     },
                     {
                         "Action":  "Allow",
                         "Direction":  "In",
                         "Id":  "profile-kns.default-aBMQCbsUMESPKGRp-0",
                         "InternalPort":  0,
                         "LocalAddresses":  "",
                         "LocalPort":  0,
                         "Priority":  1000,
                         "Protocol":  256,
                         "RemoteAddresses":  "",
                         "RemotePort":  0,
                         "RuleType":  "Switch",
                         "Scope":  0,
                         "ServiceName":  "",
                         "Type":  "ACL"
                     },
                     {
                         "Action":  "Block",
                         "Direction":  "In",
                         "InternalPort":  0,
                         "LocalAddresses":  "",
                         "LocalPort":  0,
                         "Priority":  1001,
                         "Protocol":  256,
                         "RemoteAddresses":  "",
                         "RemotePort":  0,
                         "RuleType":  "Switch",
                         "Scope":  0,
                         "ServiceName":  "",
                         "Type":  "ACL"
                     },
                     {
                         "Action":  "Allow",
                         "Direction":  "In",
                         "InternalPort":  0,
                         "LocalAddresses":  "",
                         "LocalPort":  0,
                         "Priority":  0,
                         "Protocol":  256,
                         "RemoteAddresses":  "",
                         "RemotePort":  0,
                         "RuleType":  "Host",
                         "Scope":  0,
                         "ServiceName":  "",
                         "Type":  "ACL"
                     },
                     {
                         "Action":  "Allow",
                         "Direction":  "Out",
                         "Id":  "profile-kns.default-8iYOzpfn3SU3eATK-0",
                         "InternalPort":  0,
                         "LocalAddresses":  "",
                         "LocalPort":  0,
                         "Priority":  1000,
                         "Protocol":  256,
                         "RemoteAddresses":  "",
                         "RemotePort":  0,
                         "RuleType":  "Switch",
                         "Scope":  0,
                         "ServiceName":  "",
                         "Type":  "ACL"
                     },
                     {
                         "Action":  "Block",
                         "Direction":  "Out",
                         "InternalPort":  0,
                         "LocalAddresses":  "",
                         "LocalPort":  0,
                         "Priority":  1001,
                         "Protocol":  256,
                         "RemoteAddresses":  "",
                         "RemotePort":  0,
                         "RuleType":  "Switch",
                         "Scope":  0,
                         "ServiceName":  "",
                         "Type":  "ACL"
                     },
                     {
                         "Action":  "Allow",
                         "Direction":  "Out",
                         "InternalPort":  0,
                         "LocalAddresses":  "",
                         "LocalPort":  0,
                         "Priority":  0,
                         "Protocol":  256,
                         "RemoteAddresses":  "",
                         "RemotePort":  0,
                         "RuleType":  "Host",
                         "Scope":  0,
                         "ServiceName":  "",
                         "Type":  "ACL"
                     }
                 ],
    "PrefixLength":  26,
    "Resources":  {
                      "AdditionalParams":  {

                                           },
                      "AllocationOrder":  12,
                      "Allocators":  [
                                         {
                                             "AdapterNetCfgInstanceId":  "{E3CFA3E0-13BB-4DE3-B17B-714A97D2268A}",
                                             "AdditionalParams":  {

                                                                  },
                                             "AllocationOrder":  0,
                                             "CompartmendId":  4,
                                             "Connected":  true,
                                             "DevicelessNic":  true,
                                             "EndpointNicGuid":  "B2CBCEC3-CB3D-4523-82CF-955042C310B0",
                                             "EndpointPortGuid":  "42672318-060C-4CB4-A6AA-99BCA41A980C",
                                             "Health":  {
                                                            "LastErrorCode":  0,
                                                            "LastUpdateTime":  132695159565504317
                                                        },
                                             "Hidden":  false,
                                             "ID":  "5949DAEF-32B5-41CE-A102-7517F20F8549",
                                             "InterfaceGuid":  "E3CFA3E0-13BB-4DE3-B17B-714A97D2268A",
                                             "IsPolicy":  false,
                                             "IsolationId":  4096,
                                             "MTU":  1450,
                                             "MacAddress":  "0E-2A-0A-01-12-43",
                                             "ManagementPort":  false,
                                             "NicFriendlyName":  "0dd146321a68d26b7d32e1bdf87ed5967296e58d97918858fb27c2ca55e3b586_Calico",
                                             "PreferredPortFriendlyName":  "89958F54-AEE8-4050-B1E1-CB12A86CE8BC",
                                             "State":  3,
                                             "SwitchId":  "B7214041-3FAB-4E73-81DD-CE76F5022228",
                                             "Tag":  "Endpoint Port",
                                             "WaitForIpv6Interface":  false,
                                             "nonPersistentPort":  false
                                         },
                                         {
                                             "AdditionalParams":  {

                                                                  },
                                             "AllocationOrder":  1,
                                             "Health":  {
                                                            "LastErrorCode":  0,
                                                            "LastUpdateTime":  132695159566604298
                                                        },
                                             "ID":  "4738EDD1-231D-4D3F-AF84-06572431FA90",
                                             "InDefaultIsolation":  false,
                                             "IsPolicy":  true,
                                             "IsolationID":  4096,
                                             "State":  3,
                                             "Tag":  "Isolation Policy"
                                         },
                                         {
                                             "AdditionalParams":  {

                                                                  },
                                             "AllocationOrder":  3,
                                             "CA":  "10.1.18.67",
                                             "Health":  {
                                                            "LastErrorCode":  0,
                                                            "LastUpdateTime":  132695159566804288
                                                        },
                                             "ID":  "6DA007F6-407F-41F9-AF5F-DC95BEF1DD9A",
                                             "IsLocal":  true,
                                             "IsPolicy":  true,
                                             "PA":  "172.16.0.65",
                                             "State":  3,
                                             "Tag":  "VNET Policy"
                                         },
                                         {
                                             "AdditionalParams":  {

                                                                  },
                                             "AllocationOrder":  4,
                                             "Health":  {
                                                            "LastErrorCode":  0,
                                                            "LastUpdateTime":  132695159566814288
                                                        },
                                             "ID":  "C99E9C1E-3E1B-45DF-A68D-DA4F55B03B2F",
                                             "IsPolicy":  true,
                                             "LocalRoutedVip":  false,
                                             "NatExceptions_0":  "10.152.183.0/24",
                                             "NatExceptions_1":  "10.1.0.0/16",
                                             "State":  3,
                                             "Tag":  "VFP OutBound NAT Policy",
                                             "TcpNatPool":  "SLB_NATPOOL_IPV4_TCP_10.1.18.67172.16.0.65C99E9C1E-3E1B-45DF-A68D-DA4F55B03B2F0",
                                             "UdpNatPool":  "SLB_NATPOOL_IPV4_UDP_10.1.18.67172.16.0.65C99E9C1E-3E1B-45DF-A68D-DA4F55B03B2F0",
                                             "VIP":  ""
                                         },
                                         {
                                             "AdditionalParams":  {

                                                                  },
                                             "AllocationOrder":  5,
                                             "DestinationPrefix":  "10.152.183.0/24",
                                             "Health":  {
                                                            "LastErrorCode":  0,
                                                            "LastUpdateTime":  132695159566834289
                                                        },
                                             "ID":  "1371C099-32A5-4D5D-BD1C-A8F092B26C3C",
                                             "IsPolicy":  true,
                                             "NeedEncap":  true,
                                             "NextHop":  "",
                                             "State":  3,
                                             "Tag":  "VFP Route Policy"
                                         },
                                         {
                                             "AdditionalParams":  {

                                                                  },
                                             "AllocationOrder":  6,
                                             "DestinationPrefix":  "172.16.0.65/32",
                                             "Health":  {
                                                            "LastErrorCode":  0,
                                                            "LastUpdateTime":  132695159566854300
                                                        },
                                             "ID":  "289DD17E-6486-4F77-AB1B-5B341F4C8BB2",
                                             "IsPolicy":  true,
                                             "NextHop":  "00-15-5D-00-1F-0F",
                                             "State":  1,
                                             "Tag":  "VFP Host Route Policy"
                                         },
                                         {
                                             "AdditionalParams":  {

                                                                  },
                                             "AllocationOrder":  9,
                                             "DestinationPrefix":  "10.1.32.0/26",
                                             "DistributedRouterMacAddress":  "66-5f-4c-f2-73-c3",
                                             "Health":  {
                                                            "LastErrorCode":  0,
                                                            "LastUpdateTime":  132695159614797739
                                                        },
                                             "ID":  "A5429390-D864-4080-97AB-27D5B11D9DE9",
                                             "IsPolicy":  true,
                                             "IsolationId":  4096,
                                             "ProviderAddress":  "172.16.0.89",
                                             "State":  1,
                                             "Tag":  "VFP Remote Subnet"
                                         },
                                         {
                                             "AdditionalParams":  {

                                                                  },
                                             "AllocationOrder":  11,
                                             "Health":  {
                                                            "LastErrorCode":  0,
                                                            "LastUpdateTime":  132695159897120400
                                                        },
                                             "ID":  "BEB637E6-292E-4CF9-9DD6-0EF79FB326C0",
                                             "IsPolicy":  true,
                                             "Rules":  [
                                                           {
                                                               "Action":  "Allow",
                                                               "Direction":  "In",
                                                               "Id":  "allow-host-to-endpoint",
                                                               "InternalPort":  0,
                                                               "LocalAddresses":  "",
                                                               "LocalPort":  0,
                                                               "Priority":  900,
                                                               "Protocol":  256,
                                                               "RemoteAddresses":  "172.16.0.65/32,172.26.160.1/32",
                                                               "RemotePort":  0,
                                                               "RuleType":  "Switch",
                                                               "Scope":  0,
                                                               "ServiceName":  "",
                                                               "Type":  "ACL"
                                                           },
                                                           {
                                                               "Action":  "Allow",
                                                               "Direction":  "In",
                                                               "Id":  "profile-kns.default-aBMQCbsUMESPKGRp-0",
                                                               "InternalPort":  0,
                                                               "LocalAddresses":  "",
                                                               "LocalPort":  0,
                                                               "Priority":  1000,
                                                               "Protocol":  256,
                                                               "RemoteAddresses":  "",
                                                               "RemotePort":  0,
                                                               "RuleType":  "Switch",
                                                               "Scope":  0,
                                                               "ServiceName":  "",
                                                               "Type":  "ACL"
                                                           },
                                                           {
                                                               "Action":  "Block",
                                                               "Direction":  "In",
                                                               "InternalPort":  0,
                                                               "LocalAddresses":  "",
                                                               "LocalPort":  0,
                                                               "Priority":  1001,
                                                               "Protocol":  256,
                                                               "RemoteAddresses":  "",
                                                               "RemotePort":  0,
                                                               "RuleType":  "Switch",
                                                               "Scope":  0,
                                                               "ServiceName":  "",
                                                               "Type":  "ACL"
                                                           },
                                                           {
                                                               "Action":  "Allow",
                                                               "Direction":  "In",
                                                               "InternalPort":  0,
                                                               "LocalAddresses":  "",
                                                               "LocalPort":  0,
                                                               "Priority":  0,
                                                               "Protocol":  256,
                                                               "RemoteAddresses":  "",
                                                               "RemotePort":  0,
                                                               "RuleType":  "Host",
                                                               "Scope":  0,
                                                               "ServiceName":  "",
                                                               "Type":  "ACL"
                                                           },
                                                           {
                                                               "Action":  "Allow",
                                                               "Direction":  "Out",
                                                               "Id":  "profile-kns.default-8iYOzpfn3SU3eATK-0",
                                                               "InternalPort":  0,
                                                               "LocalAddresses":  "",
                                                               "LocalPort":  0,
                                                               "Priority":  1000,
                                                               "Protocol":  256,
                                                               "RemoteAddresses":  "",
                                                               "RemotePort":  0,
                                                               "RuleType":  "Switch",
                                                               "Scope":  0,
                                                               "ServiceName":  "",
                                                               "Type":  "ACL"
                                                           },
                                                           {
                                                               "Action":  "Block",
                                                               "Direction":  "Out",
                                                               "InternalPort":  0,
                                                               "LocalAddresses":  "",
                                                               "LocalPort":  0,
                                                               "Priority":  1001,
                                                               "Protocol":  256,
                                                               "RemoteAddresses":  "",
                                                               "RemotePort":  0,
                                                               "RuleType":  "Switch",
                                                               "Scope":  0,
                                                               "ServiceName":  "",
                                                               "Type":  "ACL"
                                                           },
                                                           {
                                                               "Action":  "Allow",
                                                               "Direction":  "Out",
                                                               "InternalPort":  0,
                                                               "LocalAddresses":  "",
                                                               "LocalPort":  0,
                                                               "Priority":  0,
                                                               "Protocol":  256,
                                                               "RemoteAddresses":  "",
                                                               "RemotePort":  0,
                                                               "RuleType":  "Host",
                                                               "Scope":  0,
                                                               "ServiceName":  "",
                                                               "Type":  "ACL"
                                                           }
                                                       ],
                                             "State":  3,
                                             "Tag":  "ACL Policy"
                                         }
                                     ],
                      "Health":  {
                                     "LastErrorCode":  0,
                                     "LastUpdateTime":  132695159565324289
                                 },
                      "ID":  "CF68487D-F974-4782-8FEF-10FA99E27BF2",
                      "PortOperationTime":  0,
                      "State":  1,
                      "SwitchOperationTime":  0,
                      "VfpOperationTime":  0,
                      "parentId":  "3A61332E-3B33-4EEB-824E-44CFA68B09B7"
                  },
    "SharedContainers":  [
                             "0dd146321a68d26b7d32e1bdf87ed5967296e58d97918858fb27c2ca55e3b586",
                             "b57b0d437e0794207191f5c3ab68e7647687cfffc8b9996edee66141fc995989"
                         ],
    "StartTime":  132695159566594299,
    "State":  3,
    "Type":  "Overlay",
    "Version":  38654705669,
    "VirtualNetwork":  "683BE31B-EAAA-4D0C-A696-B800434252E3",
    "VirtualNetworkName":  "Calico"
}
```

**Comment by liquidtoon01** [comment]
I think I may have whittled it down to it being the VXLAN Tunnel that is the issue but I'm not sure how to troubleshoot beyond this?

**Comment by song-jiang** [comment]
Can you check node configuration section of config.ps1 and confirm $env:NodeName is set to Kubernetes node name?
```
The NODENAME variable should be set to match the Kubernetes Node name of this host.
```

**Comment by wtfacoconut** [comment]
Hello. I too am experiencing a similar (maybe the same) issue.
In case it helps, these were the commands I ran to get a windows worker up and running with calico. Everything below was run from an admin powershell prompt:
```
mkdir C:\k\
# I then transferred a copy of my kube config file to c:\k\
Invoke-WebRequest https://docs.projectcalico.org/scripts/install-calico-windows.ps1 -OutFile C:\install-calico-windows.ps1
C:\install-calico-windows.ps1 -DownloadOnly yes -KubeVersion 1.19.3
C:\CalicoWindows\install-calico.ps1
C:\CalicoWindows\kubernetes\install-kube-services.ps1
Start-Service kubelet
Start-Service kube-proxy
```
After doing so, my windows pods are experiencing the same issues as @liquidtoon01 .

**Comment by lmm** [comment]
I'm not sure what's going on. One similarity is that the two issues are both on microk8s. I think I'll need to read a bit more about microk8s and maybe try to repro this.

**Comment by song-jiang** [comment]
Can you check if `$env:NODENAME = $(hostname).ToLower()` has the same node name value as you run `kubectl get node`?

**Comment by slavina-rumenova** [comment]
I'm having the same issue on microK8s with one CentOs7 & one Win19 node (only difference is the IP pool that I am using). Can confirm that $env:NODENAME = $(hostname).ToLower() is the same as the node name:

`$(hostname).ToLower()
win-t8mcams9iui`

```
microk8s kubectl get nodes
NAME                       STATUS   ROLES    AGE     VERSION
zrh-centos7-tst   Ready    <none>   25h     v1.21.6-3+dd57cd4fdc581a
win-t8mcams9iui            Ready    <none>   3h42m   v1.21.6
```

**Comment by lmm** [comment]
Thanks @slavina-rumenova for confirming. I will try to repro this on microK8s and report back.

**Comment by song-jiang** [comment]
@slavina-rumenova Could you help to zip and attach contents of `c:\CalicoWindows\logs`? What is the Windows OS version you are using?

**Comment by slavina-rumenova** [comment]
@song-jiang I'm running Windows Server 2019 Standard v 1809.
Here are the calico logs:
[calicologs.zip](https://github.com/projectcalico/calico/files/7633909/calicologs.zip)

**Comment by lmm** [comment]
@slavina-rumenova the calico-node service is stuck at:


> 2021-12-01 04:30:51.183 [ERROR][6804] startup/startup.go 115: Unable to ensure datastore is migrated. error=unable to query ClusterInformation to determine Calico version: Get "https://10.93.22.20:16443/apis/crd.projectcalico.org/v1/clusterinformations/default": dial tcp 10.93.22.20:16443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

Can you check that your Windows node can reach the 10.93.22.20 on tcp port 16443?

**Comment by lmm** [comment]
> Thanks @slavina-rumenova for confirming. I will try to repro this on microK8s and report back.

I was able to bring up a microk8s cluster with a Calico for Windows node. I also ran into the same error log above. After resolving the connectivity issue, I reinstalled Calico for Windows and confirmed pod to pod connectivity with some test Linux and Windows pods.

(I found this reference on microk8s ports helpful: https://microk8s.io/docs/services-and-ports)
